lib/api/CDataFrameTrainBoostedTreeClassifierRunner.cc

```diff
@@ -68,6 +68,13 @@ CDataFrameTrainBoostedTreeClassifierRunner::CDataFrameTrainBoostedTreeClassifier
                   this->dependentVariableFieldName()) == categoricalFieldNames.end()) {
         HANDLE_FATAL(<< "Input error: trying to perform classification with numeric target.");
     }
+    const std::set<std::string> predictionFieldNameBlacklist{
+        IS_TRAINING_FIELD_NAME, PREDICTION_PROBABILITY_FIELD_NAME, TOP_CLASSES_FIELD_NAME};
+    if (predictionFieldNameBlacklist.count(this->predictionFieldName())) {
```

Comment:
I'd prefer an explicit comparison here, e.g `if (predictionFieldNameBlacklist.count(this->predictionFieldName()) > 0)`
Looking through our code there's no consistency but the bulk of such comparisons are explicit.
Incidentally there's nothing about this in the `STYLEGUIDE` either, maybe there should be?

Reply:
We used to have this in the old Word doc coding standards - section 5.27: "if statements SHOULD be fully qualified".  It got lost when that was converted to markdown.

Reply:
Created #862 to add this to the `STYLEGUIDE`

Reply:
> I'd prefer an explicit comparison here, e.g `if (predictionFieldNameBlacklist.count(this->predictionFieldName()) > 0)`

Done.